### PR TITLE
feat: light-weighted dev sdist builder

### PR DESCRIPTION
**Pull request by frostming**
Signed-off-by: Frost Ming <me@frostming.com>

## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml/_internal/bento/install.sh.j2 [hunk 1]
```diff
@@ -14,6 +14,7 @@ REQUIREMENTS_LOCK="$BASEDIR/requirements.lock.txt"
 WHEELS_DIR="$BASEDIR/wheels"
 BENTOML_VERSION=${BENTOML_VERSION:-<< bentoml_version >>}
 # Install python packages, prefer installing the requirements.lock.txt file if it exist
+pushd "$BASEDIR"
```
Comment (aarnphm) [inline src/bentoml/_internal/bento/install.sh.j2:17]:
```suggestion
pushd "$BASEDIR" &>/dev/null
```

File: src/bentoml/_internal/bento/install.sh.j2 [hunk 2]
```diff
@@ -23,6 +24,7 @@ else
         pip3 install -r "$REQUIREMENTS_TXT" "${PIP_ARGS[@]}"
     fi
 fi
+popd
```
Comment (aarnphm) [inline src/bentoml/_internal/bento/install.sh.j2:27]:
```suggestion
popd &>/dev/null
```

**Review by aarnphm — APPROVED** [review]

**Comment by aarnphm** [comment]
Will update for openllm as well 😄

**Comment by frostming** [comment]
> Will update for openllm as well 😄
> 
> actually can we also remove the build dependency?

Unfortunately `build` is a dependency of `pip-tools`.

**Comment by aarnphm** [comment]
> > Will update for openllm as well 😄
> > actually can we also remove the build dependency?
> 
> Unfortunately `build` is a dependency of `pip-tools`, but at least we can remove it from the test depenencies

Oh I mean we can remove build from our deps. build as a transitive is fine since pip-tools will manage it.

**Pull request merged**